server/src/main/java/org/candlepin/service/impl/HypervisorUpdateAction.java

```diff
@@ -90,115 +101,150 @@ public Result update(
+                    jobReporterId);
+            }
+            boolean typeUpdated = false;
+            if (!hypervisorType.getId().equals(resultHost.getTypeId())) {
+                typeUpdated = true;
+                resultHost.setType(hypervisorType);
+            }
 
-                final boolean factsUpdated = consumerResource.checkForFactsUpdate(knownHost, incoming);
+            final GuestMigration guestMigration = new GuestMigration(consumerCurator)
+                .buildMigrationManifest(incomingHost, resultHost);
 
-                if (factsUpdated || guestMigration.isMigrationPending() || typeUpdated ||
-                    hypervisorIdUpdated) {
-                    knownHost.setLastCheckin(new Date());
-                    guestMigration.migrate(false);
-                    result.addUpdated(this.translator.translate(knownHost, HypervisorConsumerDTO.class));
-                }
-                else {
-                    result.addUnchanged(
-                        this.translator.translate(knownHost, HypervisorConsumerDTO.class));
-                }
+            final boolean factsUpdated = consumerResource.checkForFactsUpdate(resultHost, incomingHost);
+
+            if (factsUpdated || guestMigration.isMigrationPending() || typeUpdated ||
+                hypervisorIdUpdated || nameUpdated) {
+                resultHost.setLastCheckin(new Date());
+                guestMigration.migrate(false);
+                result.addUpdated(this.translator.translate(resultHost, HypervisorConsumerDTO.class));
+            }
+            else {
+                result.addUnchanged(
+                    this.translator.translate(resultHost, HypervisorConsumerDTO.class));
             }
+
             // update reporter id if it changed
-            if (jobReporterId != null && reportedOnConsumer != null &&
-                reportedOnConsumer.getHypervisorId() != null &&
-                (reportedOnConsumer.getHypervisorId().getReporterId() == null ||
-                !jobReporterId.contentEquals(reportedOnConsumer.getHypervisorId().getReporterId()))) {
-                reportedOnConsumer.getHypervisorId().setReporterId(jobReporterId);
+            if (jobReporterId != null && resultHost != null &&
+                resultHost.getHypervisorId() != null &&
+                (resultHost.getHypervisorId().getReporterId() == null ||
+                !jobReporterId.contentEquals(resultHost.getHypervisorId().getReporterId()))) {
+                resultHost.getHypervisorId().setReporterId(jobReporterId);
+            }
+            try {
+                consumerCurator.update(resultHost);
             }
-            else if (jobReporterId == null) {
-                log.debug("hypervisor checkin reported asynchronously without reporter id " +
-                    "for hypervisor:{} of owner:{}", hypervisorId, ownerKey);
+            catch (Exception e) {
```

Comment:
Same thing here -- we need something to trigger a rollback when this exception occurs.